omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -1425,16 +1450,34 @@ def _get_head_goal_q(self, target_obj_pose):
 
     def _empty_action(self):
         """
-        Get a no-op action that allows us to run simulation without changing robot configuration.
+        Generate a no-op action that will keep the robot still but aim to move the arms to the saved pose targets, if possible
 
         Returns:
             th.tensor or None: Action array for one step for the robot to do nothing
         """
         action = th.zeros(self.robot.action_dim)
         for name, controller in self.robot._controllers.items():
+            # if desired arm targets are available, generate an action that moves the arms to the saved pose targets
+            if name in self._arm_targets:
+                if isinstance(controller, JointController):
+                    target_joint_pos = self._arm_targets[name]
+                    current_joint_pos = self.robot.get_joint_positions()[self._manipulation_control_idx]
+                    if controller.use_delta_commands:
+                        partial_action = target_joint_pos - current_joint_pos
+                    else:
+                        partial_action = target_joint_pos
+                else:
+                    target_pose = self._arm_targets[name]
+                    target_orn_axisangle = target_pose[1]
+                    current_pose = self._get_pose_in_robot_frame(
+                        (self.robot.get_eef_position(self.arm), self.robot.get_eef_orientation(self.arm))
+                    )
+                    delta_pos = target_pose[0] - current_pose[0]
+                    partial_action = th.cat([delta_pos, target_orn_axisangle])
+            else:
+                partial_action = controller.compute_no_op_action(self.robot.get_control_dict())
             action_idx = self.robot.controller_action_idx[name]
-            no_op_action = controller.compute_no_op_action(self.robot.get_control_dict())
-            action[action_idx] = no_op_action
+            action[action_idx] = partial_action
```

Comment:
Failing test suggests this can be a tuple sometimes? That's concerning. Lets find out why.